R/dots.R

```diff
@@ -1,23 +1,70 @@
-#' Extract dots with splicing semantics
+#' Extract tidy dots
 #'
-#' These functions evaluate all arguments contained in `...` and
-#' return them as a list. They both splice their arguments if they
-#' qualify for splicing. See [ll()] for information about splicing
-#' and below for the kind of arguments that qualify for splicing.
+#' @description
 #'
-#' `dots_list()` has _explicit splicing semantics_: it splices lists
-#' that are explicitly marked for [splicing][ll] with the
-#' [splice()] adjective. `dots_splice()` on the other hand has _list
-#' splicing semantics_: in addition to lists marked explicitly for
-#' splicing, [bare][is_bare_list] lists are spliced as well.
+#' `dots_list(...)` is equivalent to `list(...)` but provides tidy
+#' dots semantics:
+#'
+#' - You can splice other lists with the
+#'   [unquote-splice][quasiquotation] `!!!` operator.
+#'
+#' - You can unquote names by using the [unquote][quasiquotation]
+#'   operator `!!` on the left-hand side of `:=`.
+#'
+#' We call quasiquotation support in dots **tidy dots** semantics and
+#' functions taking dots with `dots_list()` tidy dots functions.
+#' Quasiquotation is an alternative to `do.call()` idioms and gives
+#' the users of your functions an uniform syntax to supply a variable
+#' number of arguments or a variable name.
+#'
+#' Note that while all tidy eval [quoting functions][quotation] have
+#' tidy dots semantics, not all tidy dots functions are quoting
+#' functions. `dots_list()` is for standard functions, not quoting
+#' functions.
+#'
+#'
+#' @section Comparison to tidy eval splicing:
+#'
+#' The `!!!` operator works differently in _standard_ functions taking
+#' dots with `dots_list()` than in _quoting_ functions taking dots
+#' with [enexprs()] or [enquos()].
+#'
+#' * In quoting functions `!!!` disaggregates its argument (let's call
+#'   it `x`) into as many objects as there are elements in
+#'   `x`. E.g. `quo(foo(!!! c(1, 2)))` is completely equivalent to
+#'   `quo(foo(1, 2))`. The creation of those separate objects has an
+#'   overhead but is typically not important when manipulating calls
+#'   because function calls typically take a small number of
+#'   arguments.
+#'
+#' * In standard functions, disaggregating the spliced collection
+#'   would have a negative performance impact in cases where
+#'   `dots_list()` is used to build up data structures from user
+#'   inputs. To avoid this spliced inputs are marked with [splice()]
+#'   and the final list is built with (the equivalent of)
+#'   `flatten_if(dots, is_spliced)`.
 #'
 #'
 #' @section Life cycle:
 #'
-#' * `dots_splice()` is experimental. It is part of our experiments
-#'   with dots semantics. We now lean towards adopting a single type
-#'   of dots semantics (those of `dots_list()`) where splicing is
-#'   explicit.
+#' * `dots_list()` returns named dots, even if no names were supplied.
+#'   This behaviour is for consistency with dots returned by [exprs()]
```

Comment:
Isn't it inconsistent? `exprs()` has `.named = FALSE`

Reply:
`.named = FALSE` still names the output with an empty vector of names, we should probably tweak the UI to make it more explicit.